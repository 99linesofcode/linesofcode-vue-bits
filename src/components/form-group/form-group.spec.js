import { shallowMount } from '@vue/test-utils';

import FormGroup from './form-group';

const wrapperFactory = propsData => {
    return shallowMount(FormGroup, {
        propsData: {
            label: 'First name',
            name: 'first_name',
            error: 'First name is required.',
            ...propsData
        },
        slots: {
            default: '<div class="mock-slot"></div>'
        }
    });
};

describe('FormGroup.vue', () => {
    it('matches the previously stored snapshot', () => {
        const wrapper = wrapperFactory();

        expect(wrapper).toMatchSnapshot();

        wrapper.destroy();
    });

    describe('label prop', () => {
        it('renders a label element when truthy', () => {
            const wrapper = wrapperFactory();

            expect(wrapper.find('label').text()).toEqual(wrapper.vm.label);

            wrapper.destroy();
        });

        it('hides the label element when falsy', () => {
            const wrapper = wrapperFactory({ label: null });

            expect(wrapper.find('label').exists()).toBe(false);

            wrapper.destroy();
        });
    });

    describe('error prop', () => {
        it('renders an error message when truthy', () => {
            const wrapper = wrapperFactory();

            expect(wrapper.find('span').text()).toEqual(wrapper.vm.error);

            wrapper.destroy();
        });

        it('hides the error message when falsy', () => {
            const wrapper = wrapperFactory({ error: null });

            expect(wrapper.find('span').exists()).toBe(false);

            wrapper.destroy();
        });
    });

    describe('when passed child content', () => {
        it('renders as the default slot', () => {
            const wrapper = wrapperFactory();

            expect(wrapper.findAll('.mock-slot').length).toBe(1);

            wrapper.destroy();
        });
    });
});
