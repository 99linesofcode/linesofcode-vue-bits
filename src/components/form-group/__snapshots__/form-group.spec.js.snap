// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup.vue matches the previously stored snapshot 1`] = `
<div class="mt-4"><label for="first_name" class="text-xs uppercase">First name</label>
  <div class="mock-slot"></div> <span class="text-red-400 text-xs">First name is required.</span>
</div>
`;
